### DISCUSSION TITLE
How to access /storage/emulated/0/Android/data ?

### DISCUSSION THREAD

**Discussion by Fancyflame** [termux/termux-app#2398] [Q&A] [answered]
I have looked similar issues in this repository before,  which has not been settled. I know that due to the restriction of Android 10+, Termux cannot access to it directly, but, it  indeed have another ways to gain the permission without using bugs, just like ES file browser do.

![Screenshot_2021-11-19-13-36-50-716_com estrongs android pop](https://user-images.githubusercontent.com/31250625/142571719-4d55df30-e445-4282-b9ef-d8f524328a8a.jpg)

That way is to popup the file browser of system, ask the user to browse the Android/data directory and select all the file, then the caller APP could gain the permission permenantly. The same way can be used to gain the access extension SD card as well.

So, could Termux gain the access to Android/data as ES File Browser do?

**Comment by agnostic-apollo** [comment, marked as answer]
That prompt gets SAF access for the `Android/data` directory and only `SAF` APIs can be used to get the files, you won't get access in shell (unless wrappers are written by someone). And that hack is gone in android 12 anyways as far as I know.

https://github.com/termux/termux-app/issues/2196#issuecomment-890325045

https://wiki.termux.com/wiki/Internal_and_external_storage

**Comment by Esensats** [comment]
Is it possible to use Shizuku for this?

**Reply by TheKingFireS** [reply]
Yes, press "Use Shizuku in terminal apps" and read tutorial. It allow use of Termux tools, that's how I able to download files straight into `Android/data/com.example/files` folder.

**Reply by Fractal420** [reply]
It's not perfect solution as you need to enter shell and there is no additional scripts working there like command hits or anything else.
